### ISSUE TITLE
Provide license

### ISSUE THREAD

**Issue by orhun** [zesterer/funkicrab#2]
Hey,
Are you planning to add a open source license to the project soon?

**Comment by zesterer** [comment]
I'll do that at some point. I'll probably go for WTFPL or something similar. I don't really care who uses this project or for what, it was mostly just a bit of fun.

**Comment by orhun** [comment]
Yeah I can tell :) Just saying because I'm currently maintaining this project on AUR: https://aur.archlinux.org/packages/funkicrab-git/

**Comment by zesterer** [comment]
Oh, nice! I should probably make it emit an actual binary instead of just C then...

**Comment by thaliaarchi** [comment]
If you have a chance sometime, could you add a license? I'd like to use some of these heuristics in a compiler (attributed, of course).

**Comment by zesterer** [comment]
I can add a license if you want to borrow some of the code itself. That said, If you were instead looking to reimplement some of the optimisations yourself, you're completely free from license obligations: copyright only applies to the specific instance of the code that I have here. That aside, many of the optimisations implemented here are also documented elsewhere on the web!

**Comment by thaliaarchi** [comment]
I'm no expert on licensing, but I thought that reimplementing algorithms, using specific source code as reference, would be a derivative work. In any case, it sounds like you're open to the code being used in other projects, which is great and is good enough for me.
